### ISSUE TITLE
@types/gapi.client.discovery@1.0.9: use @types/gapi.client.discovery-v1 instead

### ISSUE THREAD

**Issue by tim-tribble-ai** [Maxim-Mazurok/google-api-typings-generator#920]
Hi!

Running into the same deprecation warning as mentioned here:
https://github.com/Maxim-Mazurok/google-api-typings-generator/issues/885

```
$ npm install --save-dev @maxim_mazurok/gapi.client.drive-v3
npm WARN deprecated @types/gapi.client.discovery@1.0.9: use @types/gapi.client.discovery-v1 instead; see https://github.com/Maxim-Mazurok/google-api-typings-generator/issues/652 for details
```

Seems like the package.json still reference the old 1.0.9 version?
![image](https://github.com/Maxim-Mazurok/google-api-typings-generator/assets/133814265/40b7f983-d320-491f-9339-27a6b2984c5d)

Or...not if there another way I should be adding this package?

Thanks!

**Comment by Maxim-Mazurok** [comment]
Hello, and thanks for reporting this!
TL;DR - I published [@maxim_mazurok/gapi.client.discovery-v1](https://www.npmjs.com/package/@maxim_mazurok/gapi.client.discovery-v1/v/0.1.20200806), which is a dependency of all `@types/gapi.client.*` packages, so next time you do `npm update` the warning should be gone.

Also, it's better to use `@types/gapi.client.drive-v3`, which acts as a [proxy](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fa2da0b968cd107dd1aa95c3175123115c19c13b/types/gapi.client.drive-v3/package.json#L4) to the latest `@maxim_mazurok/gapi.client.drive-v3`. I think it's better because `@types/*` is added in `types` of `tsconfig` by default.

---

When generating `@maxim_mazurok/gapi.client.discovery-v1` package - it uses the [same template](https://github.com/Maxim-Mazurok/google-api-typings-generator/blob/c05129cfa3bbc5be5d2f1d15b31b444fdf889792/src/template/package-json.dot#L18C43-L18C43) as all other packages, in which we list `@types/gapi.client.discovery-v1` as a dependency. And `@types/gapi.client.discovery-v1` depends on `@maxim_mazurok/gapi.client.discovery-v1`, so it's kinda circular dependency, but npm seems to deal fine with that.

Also, you need to know that we only publish new version of the `@maxim_mazurok/gapi.client.*` types packages when the revision number goes up (as reported by google in the API schema definition).

So even though we have updated the template to use `discovery-v1` package instead of deprecated `discovery`, there are still some packages that didn't have revision updates from google since then, and so new versions were never released.

I have created an [exception](https://github.com/Maxim-Mazurok/google-api-typings-generator/blob/c05129cfa3bbc5be5d2f1d15b31b444fdf889792/src/utils.ts#L289) for `discovery-v1` and released it manually, so we should get rid of this warning for most popular APIs.

However, there are still some types that use the deprecated `discovery` package:
![image](https://github.com/Maxim-Mazurok/google-api-typings-generator/assets/7756211/c7c2e2e6-32c1-47a9-b962-edb838115c5b)
They all seem to be relatively unpopular I think. Also I believe there should be no real difference between `discovery` and `discovery-v1`, so I won't be manually updating those for now, unless there's a demand/issue.

Hope this explains it :)
